File: parnamondal/Week - 2/Day-4/Reverse Bits.py
class Solution:
    # @param A : unsigned integer
    # @return an unsigned integer
    
    	# Time Complexity - 0(1)
        # Space Complexity - O(1)
    
    def reverse(self, a):
        if a==0:
            return 0
        no=bin(a)[2:]
        f=len(no)
        h=32-f
        s='0'*h+no
        s=s[::-1]
        return int(s,2)
            

File: parasmittal/DAY 12/Reverse Bits.cpp
unsigned int Solution::reverse(unsigned int A) {
    unsigned int ans=0;
    int j=31;
    for(int i=0;i<32;i++){
        int val=(1<<i) & A;
       
        if(val)
        ans|=((1<<j));
        j--;
        
    }
   
    return ans;
}
